exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -193,6 +193,24 @@ func (m *AzureManagedControlPlane) ValidateUpdate(oldRaw runtime.Object, client
 		}
 	}
 
+	if old.Spec.ControlPlaneEndpoint.Host != "" {
```

Comment:
for AzureManagedControlPlane is there is no "ready" equivalent to make sure the user doesn't set it on the initial create? If so that's unfortunate, since that's where the user would be most likely to set it in the first place since that's where they set all other configurations

Reply:
There isn't actually. Well, the resource spec includes that field but we're not setting it.

Reply:
Would we want to address this gap in this PR? Something like this would do the trick

```
$ git diff
diff --git a/azure/scope/managedcontrolplane.go b/azure/scope/managedcontrolplane.go
index 251d77c59..4d1a01579 100644
--- a/azure/scope/managedcontrolplane.go
+++ b/azure/scope/managedcontrolplane.go
@@ -532,6 +532,7 @@ func (s *ManagedControlPlaneScope) GetAllAgentPoolSpecs() ([]azure.ResourceSpecG
 
 // SetControlPlaneEndpoint sets a control plane endpoint.
 func (s *ManagedControlPlaneScope) SetControlPlaneEndpoint(endpoint clusterv1.APIEndpoint) {
+       s.ControlPlane.Status.Ready = true
        if s.ControlPlane.Spec.ControlPlaneEndpoint.Host == "" {
                s.ControlPlane.Spec.ControlPlaneEndpoint.Host = endpoint.Host
        }
diff --git a/exp/api/v1beta1/azuremanagedcontrolplane_webhook.go b/exp/api/v1beta1/azuremanagedcontrolplane_webhook.go
index 451df1c78..a81a0185c 100644
--- a/exp/api/v1beta1/azuremanagedcontrolplane_webhook.go
+++ b/exp/api/v1beta1/azuremanagedcontrolplane_webhook.go
@@ -90,6 +90,21 @@ func (m *AzureManagedControlPlane) ValidateCreate(client client.Client) error {
                        "can be set only if the AKS feature flag is enabled",
                )
        }
+       if !m.Status.Ready {
+               errorMessage := "can not be set by the user, will be set automatically by AKS after the cluster is Ready"
+               if m.Spec.ControlPlaneEndpoint.Host != "" {
+                       return field.Forbidden(
+                               field.NewPath("Spec", "ControlPlaneEndpoint", "Host"),
+                               errorMessage,
+                       )
+               }
+               if m.Spec.ControlPlaneEndpoint.Port != 0 {
+                       return field.Forbidden(
+                               field.NewPath("Spec", "ControlPlaneEndpoint", "Port"),
+                               errorMessage,
+                       )
+               }
+       }
        return m.Validate(client)
 }
 
@@ -193,7 +208,7 @@ func (m *AzureManagedControlPlane) ValidateUpdate(oldRaw runtime.Object, client
                }
        }
 
-       if old.Spec.ControlPlaneEndpoint.Host != "" {
+       if old.Spec.ControlPlaneEndpoint.Host != "" || !m.Status.Ready {
                if err := webhookutils.ValidateImmutable(
                        field.NewPath("Spec", "ControlPlaneEndpoint", "Host"),
                        old.Spec.ControlPlaneEndpoint.Host,
@@ -202,7 +217,7 @@ func (m *AzureManagedControlPlane) ValidateUpdate(oldRaw runtime.Object, client
                }
        }
 
-       if old.Spec.ControlPlaneEndpoint.Port != 0 {
+       if old.Spec.ControlPlaneEndpoint.Port != 0 || !m.Status.Ready {
                if err := webhookutils.ValidateImmutable(
                        field.NewPath("Spec", "ControlPlaneEndpoint", "Port"),
                        old.Spec.ControlPlaneEndpoint.Port,
```

Reply:
cc @nojnhuh

Reply:
For AzureCluster we do it right after setting the endpoint (before that data gets persisted and the webhook is called) https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azurecluster_controller.go#L253, we could do something similar here.

> Would we want to address this gap in this PR?

A separate PR is probably better since this is a bit of a scope creep, we can hold this one until it gets merged.

Reply:
I was wrong about that!

Can address the webhook readiness condition in this PR.

Reply:
O.K., I did more investigation here and it doesn't appear we can (easily) enforce Status.Ready=true as a requirement. Status=True and ControlPlaneEndpoint data populations happen during the same reconciliation, and it appears in practice the webhooks are engaged with a `m *AzureManagedControlPlane` context (and the same for `r *AzureManagedCluster` in that webhook) that does not have the Status=True data. So if we add a requirement in the webhook that says "only allow a `"` --> `"<actual endpoint data>"` update if Status=True it will always fail, thus short-circuiting the reconciliation, thus blocking the entire cluster creation flow.

I'd be curious if I'm overlooking things.